pcapng/flags.py

```diff
@@ -0,0 +1,231 @@
+"""
+Module to wrap an integer in bitwise flag/field accessors.
+"""
+
+from collections import OrderedDict
+
+from pcapng.utils import Iterable, namedtuple
+
+
+class FlagBase(object):
+    """\
+    Base class for flag types to be used in a Flags object.
+    Handles the bitwise math so subclasses don't have to worry about it.
+    """
+
+    __slots__ = [
+        "owner",
+        "offset",
+        "size",
+        "extra",
+        "mask",
+    ]
+
+    def __init__(self, owner, offset, size, extra=None):
+        if size < 1:
+            raise TypeError("Flag must be at least 1 bit wide")
+        if size > owner._nbits:
+            raise TypeError("Flag must fit into owner size")
+        self.owner = owner
+        self.offset = offset
+        self.size = size
+        self.extra = extra
+        self.mask = ((1 << self.size) - 1) << self.offset
+
+    def get_bits(self):
+        return (self.owner._value & self.mask) >> self.offset
+
+    def set_bits(self, val):
+        val &= (1 << self.size) - 1
+        self.owner._value &= ~self.mask
+        self.owner._value |= val << self.offset
+
+
+class FlagBool(FlagBase):
+    """Object representing a single boolean flag"""
+
+    def __init__(self, owner, offset, size, extra=None):
+        if size != 1:
+            raise TypeError(
+                "{cls} can only be 1 bit in size".format(cls=self.__class__.__name__)
+            )
+        super(FlagBool, self).__init__(owner, offset, size)
+
+    def get(self):
+        return bool(self.get_bits())
+
+    def set(self, val):
+        self.set_bits(int(bool(val)))
+
+
+class FlagUInt(FlagBase):
+    """\
+    Object representing an unsigned integer of the given size stored in
+    a larger bitfield
+    """
+
+    def get(self):
+        return self.get_bits()
+
+    def set(self, val):
+        self.set_bits(val)
+
+
+class FlagEnum(FlagBase):
+    """\
+    Object representing a range of values stored in part of a larger
+    bitfield
+    """
+
+    def __init__(self, owner, offset, size, extra=None):
+        if not isinstance(extra, Iterable):
+            raise TypeError(
+                "{cls} needs an iterable of values".format(cls=self.__class__.__name__)
+            )
+        extra = list(extra)
+        if len(extra) > 2 ** size:
+            raise TypeError(
+                "{cls} iterable has too many values (got {got}, "
+                + "{size} bits only address {max})".format(
```

Comment:
Does the string not fit in the same line with black?

Reply:
Nope. At the current indentation level, the end of the word 'address' is on column 88.